src/engraving/libmscore/chord.cpp

```diff
@@ -1425,34 +1425,37 @@ int Chord::calcMinStemLength()
             outsideStaffOffset = line - staff()->lines(tick()) * 4 + 4;
         }
         minStemLength += (outSidePadding + qMax(noteSidePadding, outsideStaffOffset));
-    }
 
-    if (_hook) {
-        int hookOffset = (_hook->height() - qAbs(_hook->smuflAnchor().y())) / _spatium * 4 - 2;
-        // TODO: when the SMuFL metadata includes a cutout for flags, replace this with that metadata
-        // https://github.com/w3c/smufl/issues/203
-        int cutout = up() ? 6 : 8;
-        if (beams() >= 2) {
-            cutout -= 2;
-        }
-        if (score()->styleB(Sid::useStraightNoteFlags)) {
-            cutout = 0;
-        }
-        if (minStemLength < cutout) {
-            minStemLength = hookOffset;
-        } else {
-            minStemLength += hookOffset - cutout;
+        if (_hook) {
+            double smuflAnchor = _hook->smuflAnchor().y() * (_up ? 1 : -1);
+            int hookOffset = (_hook->height() + smuflAnchor) / _spatium * 4 - 2;
+            // TODO: when the SMuFL metadata includes a cutout for flags, replace this with that metadata
+            // https://github.com/w3c/smufl/issues/203
+            int cutout = up() ? 4 * _relativeMag : 6 * _relativeMag;
```

Comment:
Are you sure that `cutout` must be an int and not a double? Or should it maybe be rounded? (since `_relativeMag` is a double)

Reply:
Yeah. The calculations are all integers because they represent quantities of 0.25sp, so 1 = 0.25sp, 2 = 0.5, etc.

The fact that all of the calculations need integers and `_relativeMag` was a double was one of the biggest challenges with this one. This line can be changed to `floor(...)` instead of just casting to integer (though they're positive anyway). I'll change it to be explicitly floored but cutout not being a double would just push the issue further down the line when it is subtracted from `hookOffset` or added to `minStemLength` or whatever.